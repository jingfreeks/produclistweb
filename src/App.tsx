import React from "react";
import { ProductDetails, Product } from "./screens";
import { Routes, Route, Navigate } from "react-router-dom";

function App() {
  return (
    <Routes>
      <Route path="/" element={<Product />} />
      <Route path="/:prodId" element={<ProductDetails />} />
    </Routes>
  );
}

export default App;
