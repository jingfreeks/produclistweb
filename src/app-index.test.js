import { renderWithProviders } from "./slice/testframework";
import App from "./index";
jest.mock("react-router-dom", () => {
  return {
    Routes: () => jest.fn(),
    Route: () => jest.fn(),
    Link:()=>jest.fn()
  };
});
jest.mock('react-dom/client',()=>{
    return{
        createRoot:()=>{
            return{
                render:jest.fn(),
                unmount:jest.fn()
            }
        }
    }
})
// Mock useRouter:
describe("App index ", () => {
  it("renders Layout unchanged to match snapshot", () => {
    const { container } = renderWithProviders(<App />);
    expect(container).toMatchSnapshot();
  });
});
