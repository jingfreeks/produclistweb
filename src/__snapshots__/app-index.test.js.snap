// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App index  renders Layout unchanged to match snapshot 1`] = `<div />`;
